### ISSUE TITLE
CA1508 False positive

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#4386]
### Analyzer

**Diagnostic ID**: CA1508

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Steps To Reproduce

````csharp
namespace ConsoleApp4
{
    public static class Program
    {
        public static bool Repro(object? input)
        {
            if (input is string s)
                return string.IsNullOrWhiteSpace(s); // warning CA1508: 'string.IsNullOrWhiteSpace(s)' is always 'false'. Remove or refactor the condition(s) to avoid dead code.
            return false;
        }
    }
}
````

### Expected behavior

No diagnostic because the result is not always false. For instance `Repro("   ")` would return true.

### Actual behavior

warning CA1508: 'string.IsNullOrWhiteSpace(s)' is always 'false'. Remove or refactor the condition(s) to avoid dead code.

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/3685
